app.R

```diff
@@ -34,33 +34,33 @@ ui <- fluidPage(theme = shinytheme('flatly'),
 )
 
 # render UI for a given subexperiment
-render_subexp_ui <- function(subexp_name) {
+render_subexp_ui <- function(subexp_name, exp_name) {
   
   tabPanel(subexp_name,
     
     sidebarPanel(class = 'push-down',
-      uiOutput(paste0('variable_select_', subexp_name)),
-      uiOutput(paste0('cultivar_select_', subexp_name))
+      uiOutput(paste0('variable_select_', exp_name, '_', subexp_name)),
+      uiOutput(paste0('cultivar_select_', exp_name, '_', subexp_name))
     ),
      
-    uiOutput(paste0(paste0('plot_hover_info_', subexp_name))),
+    uiOutput(paste0(paste0('plot_hover_info_', exp_name, '_', subexp_name))),
```

Comment:
Is `paste0(paste0(` redundant?

Reply:
That is what I thought ... but I didn't want to remove anything from the original code that wasn't causing a problem. 

I will remove the extra paste0.